src/mpi/coll/allgather/allgather.c

```diff
@@ -71,11 +71,12 @@
       verbosity   : MPI_T_VERBOSITY_USER_BASIC
       scope       : MPI_T_SCOPE_ALL_EQ
       description : >-
-        If set to true, MPI_Allgather will allow the device to override the
-        MPIR-level collective algorithms. The device still has the
-        option to call the MPIR-level algorithms manually.
-        If set to false, the device-level allgather function will not be
-        called.
+        If set to true, MPI_Allgather will allow the device to
+        override the MPIR-level collective algorithms. The device
+        still has the option to call the MPIR-level algorithms
+        manually.  If set to false, the device-level allgather
+        function will not be called.  This CVAR is only used when
+        MPIR_CVAR_DEVICE_COLLECTIVES is set to "percoll".
```

Comment:
Let me see if I get this correct: By default, all collective will call `MPID_` collective first -- because `MPIR_CVAR_DEVICE_COLLECTIVES` is set to true by default. Wait, that is a boolean cvar, what does "set to percoll" mean?

Reply:
It's not a boolean anymore.  It's an enum.

Reply:
I see. I missed that. Let me spend more time :)

Reply:
I am a top-down reader, so may I suggest --
```
When MPIR_CVAR_DEVICE_COLLECTIVES is set to "percoll", this cvar, if set to false,
skips the device-level allgather function. By default, device is allowed to override 
the MPIR-level collective algorithms. The device still has the option to call the 
MPIR-level algorithms manually.
```
-- I personally like to read the point of this variable first before reading its effect without it. :)

Reply:
Fixed.